lisa/features/gpu.py

```diff
@@ -0,0 +1,138 @@
+        cuda_repo = ""
+        linux_os: Linux = cast(Linux, self._node.os)
+
+        if not version.strip():
+            version = "10.1.105-1"
+
+        # CUDA driver installation for redhat distros
+        if isinstance(self._node.os, Redhat):
+            cuda_repo_pkg = f"cuda-repo-rhel7-{version}.x86_64.rpm"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/"
+                f"compute/cuda/repos/rhel7/x86_64/{cuda_repo_pkg}"
+            )
+            linux_os = Redhat(self._node)
+
+        # CUDA driver installation for Ubuntu distros
+        elif isinstance(self._node.os, Ubuntu):
+            release_version = self._node.os._os_version.release
+            release = re.sub("[^0-9]+", "", release_version)
+            cuda_repo_pkg = f"cuda-repo-ubuntu{release}_{version}_amd64.deb"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/compute/"
+                f"cuda/repos/ubuntu{release}/x86_64/{cuda_repo_pkg}"
+            )
+            linux_os = Ubuntu(self._node)
+
+        else:
+            raise LisaException(
+                f"Distro {self._node.os.__class__.__name__}"
+                "not supported to install CUDA driver."
+            )
+
+        # download and install the cuda driver package from the repo
+        linux_os._install_package_from_url(f"{cuda_repo}", signed=False)
+
+    def install_gpu_dep(self) -> None:
+        uname_tool = self._node.tools[Uname]
+        uname_ver = uname_tool.get_linux_information().uname_version
+
+        # install dependency libraries for redhat and CentOS
+        if isinstance(self._node.os, Redhat):
+            # install the kernel-devel and kernel-header packages
+            package_name = f"kernel-devel-{uname_ver} kernel-headers-{uname_ver}"
+            self._node.os.install_packages(package_name)
+
+            # mesa-libEGL install/update is require to avoid a conflict between
+            # libraries - bugzilla.redhat 1584740
+            package_name = "mesa-libGL mesa-libEGL libglvnd-devel"
+            self._node.os.install_packages(package_name)
+
+            # install dkms
+            package_name = "dkms"
+            self._node.os.install_packages(package_name, signed=False)
+
+        # install dependency libraraies for Ubuntu
+        elif isinstance(self._node.os, Ubuntu):
+            package_name = (
+                f"build-essential libelf-dev linux-tools-{uname_ver}"
+                f" linux-cloud-tools-{uname_ver} python libglvnd-dev ubuntu-desktop"
+            )
+            self._node.os.install_packages(package_name)
+
+    def install_compute_sdk(self, driver: ComputeSDK, version: str = "") -> None:
+        if driver == ComputeSDK.GRID:
+            self._install_grid_driver(version)
+        elif driver == ComputeSDK.CUDA:
+            self._install_cuda_driver(version)
+        else:
+            raise LisaException("No valid driver SDK name provided to install.")
```

Comment:
It looks the reason is the `driver` cannot be parsed. The message should mention it, and output the actual value of `driver`.

Reply:
Fixed